streamalert_cli/_infrastructure/modules/tf_artifact_extractor/iam.tf

```diff
@@ -0,0 +1,109 @@
+// Allow the Artifact Extractor to write to the Firehose delivering Artifacts to S3
+resource "aws_iam_role_policy" "put_artifacts_firehose" {
+  name   = "PutRecordsToArtifactsFirehose"
+  role   = var.function_role_id
+  policy = data.aws_iam_policy_document.put_artifacts_firehose_policy.json
+}
+
+data "aws_iam_policy_document" "put_artifacts_firehose_policy" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "firehose:DeleteDeliveryStream",
+      "firehose:PutRecord",
+      "firehose:PutRecordBatch",
+      "firehose:UpdateDestination"
+    ]
+
+    resources = [
+      var.function_alias_arn
+    ]
+  }
+}
+
+// IAM Role: Artifacts Firehose Delivery Stream permissions
+resource "aws_iam_role" "streamalert_kinesis_firehose" {
+  name               = "${var.prefix}_firehose_artifacts_delivery"
+  path               = "/streamalert/"
```

Comment:
thank you for paying attention :)

Reply:
😄 copy&pasta from somewhere.

Reply:
if copy/pasta is right then that's not a bad thing :P